src/frontend/sharedComponents/Map/MapView.tsx

```diff
@@ -0,0 +1,107 @@
+import Mapbox, {UserLocation} from '@rnmapbox/maps';
+import config from '../../../config.json';
+import CheapRuler from 'cheap-ruler';
+import {IconButton} from '../IconButton';
+import {LocationFollowingIcon, LocationNoFollowIcon} from '../icons';
+import {View, StyleSheet} from 'react-native';
+
+// This is the default zoom used when the map first loads, and also the zoom
+// that the map will zoom to if the user clicks the "Locate" button and the
+// current zoom is < 12.
+const DEFAULT_ZOOM = 12;
+
+const ruler = new CheapRuler(0, 'meters');
+
+Mapbox.setAccessToken(config.mapboxAccessToken);
+const MIN_DISPLACEMENT = 15;
+
+type MapViewProps = {
+  coords?: number[];
+  isFocused: boolean;
+  locationServiceEnabled?: boolean;
+};
+
+const MAP_STYLE = Mapbox.StyleURL.Outdoors;
+
+export const MapView = React.memo(
+  ({coords, locationServiceEnabled, isFocused}: MapViewProps) => {
+    const [zoom, setZoom] = React.useState(DEFAULT_ZOOM);
+    const [following, setFollowing] = React.useState(false);
+
+    React.useEffect(() => {
+      Mapbox.setTelemetryEnabled(false);
+    }, []);
+
+    return (
+      <React.Fragment>
+        <Mapbox.MapView
+          testID="mapboxMapView"
+          style={{flex: 1}}
+          logoEnabled={false}
+          pitchEnabled={false}
+          rotateEnabled={false}
+          surfaceView={true}
+          attributionPosition={{right: 8, bottom: 8}}
+          compassEnabled={false}
+          styleURL={MAP_STYLE}>
+          <Mapbox.Camera
+            defaultSettings={{
+              centerCoordinate: coords,
+              zoomLevel: zoom,
+            }}
+            centerCoordinate={following ? coords : undefined}
+            // zoomLevel={zoom}
+            animationDuration={1000}
+            animationMode="flyTo"
+            followUserLocation={false}
+          />
+          {locationServiceEnabled ? (
+            <UserLocation
+              visible={isFocused}
+              minDisplacement={MIN_DISPLACEMENT}
+            />
+          ) : null}
+        </Mapbox.MapView>
+        <View style={styles.locationButton}>
+          <IconButton onPress={() => setFollowing(prev => !prev)}>
+            {following ? <LocationFollowingIcon /> : <LocationNoFollowIcon />}
+          </IconButton>
+        </View>
+      </React.Fragment>
+    );
+  },
+  shouldComponentSkipRerender,
+);
+
+function shouldComponentSkipRerender(
+  prevProps: MapViewProps,
+  nextProps: MapViewProps,
+) {
+  // if map screen is not in focus, do not re-render
+  if (!nextProps.isFocused) return true;
+
+  if (prevProps.locationServiceEnabled !== nextProps.locationServiceEnabled)
+    return false;
+
+  if (!nextProps.coords) return true;
+
+  if (!prevProps.coords) return false;
+
+  const distanceMoved = ruler.distance(
+    [prevProps.coords[0], prevProps.coords[1]],
+    [nextProps.coords[0], nextProps.coords[1]],
+  );
+
+  if (distanceMoved < MIN_DISPLACEMENT) return true;
+
+  return false;
+}
```

Comment:
@achou11 Would you mind taking a look at this? I am trying to mimic `shouldComponentUpdate` used in a class component, like we do [here](https://github.com/digidem/mapeo-mobile/blob/f4f3350d1652f444f98df1e37d378c56da750ba7/src/frontend/sharedComponents/Map/MapView.js#L213) in Mapeo Mobile

Reply:
mind explaining the `locationServiceEnabled` check? is that needed as a prop in general?

Reply:
This seems to be the main difference between the existing implementation and what you have (although the `locationServiceEnabled` check is partially addressing that): 

https://github.com/digidem/mapeo-mobile/blob/f4f3350d1652f444f98df1e37d378c56da750ba7/src/frontend/sharedComponents/Map/MapView.js#L220-L224

not sure how meaningful that check is but just noting in case it does end up being something to include here